### PR TITLE
feat(ci): enabled automated benchmarking

### LINKED ISSUE #3245: Re-enable automatic benchmarking
This is disabled until benchmarking is working properly. We'll need to re-enable it in the future

### PR DESCRIPTION
**Pull request by maateen**
# Changes: 

- feat(ci): enabled automated benchmarking
- feat(ci): enabled automated benchmarking for pypi releases

This PR will close #3245

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Benchmarking
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*"
```
Comment (hanxiao) [inline .github/workflows/benchmark.yml:8]: maybe add also a manual button for testing and redo a test?

```suggestion
  workflow_dispatch:
  push:
    branches:
      - master
    tags:
      - "v*" 
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  push:
+    branches:
+      - master
+    tags:
+      - "v*" 
+
+jobs:
+  benchmark-tag:
+    runs-on: ubuntu-latest
+    if: ${{ GITHUB_REF != 'master' }}
```
Comment (tadejsv) [inline .github/workflows/benchmark.yml:13]: I think this won't work - you need to use the `env` context, i.e. `env.GITHUB_REF`

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - uses: benc-uk/workflow-dispatch@v1
+        with:
+          workflow: benchmark-jina
+          repo: jina-ai/jina-terraform
+          token: ${{ secrets.JINA_DEV_BOT }}
+          inputs: '{ "git_tags": ["$GITHUB_REF"], "pypi_releases": [] }'
+
+  benchmark-master:
+    runs-on: ubuntu-latest
+    if: ${{ GITHUB_REF == 'master' }}
```
Comment (tadejsv) [inline .github/workflows/benchmark.yml:24]: Samre here

**Review by tadejsv — CHANGES_REQUESTED** [review]

File: .github/workflows/benchmark.yml (hunk 2 above)
Comment (tadejsv) [reply .github/workflows/benchmark.yml:13]: You can use `github.ref` also

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (maateen) [reply .github/workflows/benchmark.yml:8]: The manual button for testing is [here](https://github.com/jina-ai/jina-terraform/actions/workflows/benchmark.yml). This workflow mainly dispatches this trigger to [jina-ai/jina-terraform](https://github.com/jina-ai/jina-terraform) repo.

I will create a redo button there as well.

**Review by hanxiao — APPROVED** [review]

**Pull request merged**